File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/gronsfeld/GronsfeldSuite.kt
package cryptobench.ciphers.gronsfeld

import cryptobench.core.Cipher
import cryptobench.core.CipherSuite
import cryptobench.core.SuiteResult
import cryptobench.verify.roundTrip

class GronsfeldSuite : CipherSuite {
    override fun name(): String = "gronsfeld"

    override fun run(): SuiteResult {
        val cipher: Cipher = GronsfeldCipher(GronsfeldKey.default())
        var passed = 0
        var failed = 0
        val start = System.nanoTime()
        for (sample in GronsfeldVectors.samples()) {
            if (roundTrip(cipher, sample)) passed++ else failed++
        }
        return SuiteResult(name(), passed, failed, System.nanoTime() - start)
    }
}

File: fixtures/scale/dart/large/lib/ciphers/gronsfeld/gronsfeld_suite.dart
import '../../core/cipher.dart';
import '../../core/cipher_suite.dart';
import '../../core/suite_result.dart';
import '../../verify/round_trip.dart';
import 'gronsfeld_cipher.dart';
import 'gronsfeld_key.dart';
import 'gronsfeld_vectors.dart';

class GronsfeldSuite implements CipherSuite {
  @override
  String name() => 'gronsfeld';

  @override
  String category() => 'cipher';

  @override
  SuiteResult run() {
    final Cipher cipher = GronsfeldCipher(GronsfeldKey.defaultKey());
    var passed = 0;
    var failed = 0;
    final start = DateTime.now();
    for (final sample in gronsfeldSamples()) {
      if (roundTripCheck(cipher, sample)) {
        passed++;
      } else {
        failed++;
      }
    }
    final elapsed = DateTime.now().difference(start).inMicroseconds;
    return SuiteResult(name(), passed, failed, elapsed);
  }
}
